Monika-After-Story/MonikaModDev
# NewO31

Adds this year's Halloween.

# Additions:
- New decor
- New costumes
- New greetings

## New methods:
- `mas_o31ShowVisuals()`
  - Shows O31 visuals
- `mas_o31HideVisuals()`
  - Hides O31 visuals, also sets the `show_vignette` global to `False` so subsequent spaceroom calls don't show it again
- `mas_isFirstSeshDay()`
  - True if today (assuming `None` is passed in) is the day of the first session

# Adjustments:
- Overhauled autoload code, now handles spritepacks with custom greet labels (defined by `greeting_o31_<spritepack_name>`
- First session on O31 does **NOT** get O31 content in the form of a costume/decor, however they do get T/T.
- Cleaned up T/T greets/fares
- S-lightning is based on doki cares too now
- Normal taking you somewhere farewell no longer provides the option of TT
- Removed a lot of persist vars which weren't really needed
- Pushed D25 season start to the 11th of December

# Testing:
- Verify things in `Adjustments` are working correctly
- Verify no other issues (typos/crashes/etc)
- Verify persists don't Sayori on Jan 7 or later
